reflex/.templates/apps/basetemplate/code/sidebar.py

```diff
@@ -0,0 +1,125 @@
+import reflex as rx
+
+from .state import State
+from .styles import *
+
+
+def sidebar_item(text: str, icon: str, url: str) -> rx.Component:
+    return rx.hstack(
+        rx.image(
+            src=icon,
+            height="2.5em",
+            padding="0.5em",
+        ),
+        rx.text(
+            text,
+        ),
+        bg=rx.cond(State.current_page == text, accent_color, "transparent"),
+        color=rx.cond(State.current_page == text, accent_text_color, text_color),
```

Comment:
i don't really like this because if the user navigates to a different URL via the address bar or clicking "Back" button in the browser, then the highlight won't also update.

i think we need to base it on the `router_data`, compared with the `url` (instead of the `text`)